### DISCUSSION TITLE
Case insensitive expression index?

### DISCUSSION THREAD

**Discussion by chuseman** [datalust/seq-tickets#2193] [Help]
Is there a way to create an expression index that is case insensitive?

I've tried to create one via `seqcli` (by specfying `--expresion "UserName ci"` for example) but after waiting a while it stays at 0 bytes. I'm assuming this is treated as an invalid/missing property.

**Comment by liammclennan** [comment]
Hi @chuseman,

You can create the index on something like `ToLower(UserName)`. 

Searches like `ToLower(UserName) = 'bob'` will then use the index. 

Expression indexes can only find expressions that are equal to a constant, so cannot be truly case insensitive.

**Reply by chuseman** [reply]
Ah okay thanks Liam. I'd expect this will be a common use case for us where we don't actually care what the case is for properties that we've indexed. I was hoping there might be some kind of magic where Seq could do this ToLower() bit behind the scenes and we wouldn't have to remember to do in every query.

**Reply by liammclennan** [reply]
@chuseman that would be nice. Seq is indexing the lower case version of the property, its just that when you want to find something in the index you have to look for the lower case version. It is not an easy problem to solve. It seems that Postgresql and MS Sql Server both rely on exactly the solution described above. Also, case sensitive search will always be faster.
